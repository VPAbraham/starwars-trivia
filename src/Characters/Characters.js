import React from 'react';
import CharacterCard from '../CharacterCard/CharacterCard';
import './Characters.css';



const Characters = (props) => {
  const characterCards = props.characters.map(character => {
    return (<li key={character.name} className='li li--character'>
      <CharacterCard
        name={character.name}
        homeworld={character.homeworld}
        homeworldPop={character.homeworld.population || 3}
        species={character.species}
        relatedFilms={character.films}
        isFavorite={false}
      />
    </li>)
  });
  return (
    <section className='section section--character-list'>
      <p>HIHIHIHIHIH</p>
      <ul className='ul ul--character-list'>
        {/* {characterCards} */}
      </ul>
    </section>
  )
}


export default Characters;