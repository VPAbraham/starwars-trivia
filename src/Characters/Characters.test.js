import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters', () => {
  let wrapper
  const addFavoriteCharacters = jest.fn()
  let characters = [{name: 'char1'}, {name: 'char2'}]
  beforeEach(() => {
    wrapper = shallow(
      <Characters
        characters={characters}
        addFav={addFavoriteCharacters}
      />
    )
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
})