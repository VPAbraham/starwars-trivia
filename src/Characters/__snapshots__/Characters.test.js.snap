// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters should match the snapshot 1`] = `
<section
  className="section section--character-list"
>
  <ul
    className="ul ul--character-list"
  >
    <li
      className="li li--character"
      key="char1"
    >
      <CharacterCard
        isFavorite={false}
        method={[MockFunction]}
        name="char1"
      />
    </li>
    <li
      className="li li--character"
      key="char2"
    >
      <CharacterCard
        isFavorite={false}
        method={[MockFunction]}
        name="char2"
      />
    </li>
  </ul>
  <div
    className="movie-crawl"
  >
    <div
      className="fade"
    />
    <section
      className="star-wars"
    >
      <div
        className="crawl"
      >
        <div
          className="title"
        >
          <p>
            Episode IV
          </p>
          <h1>
            A New Hope
          </h1>
        </div>
        <p>
          It is a period of civil war. 
          <br />
          Rebel spaceships, striking from a hidden base, 
          <br />
          have won their first victory against the evil Galactic Empire.
        </p>
        <p>
          During the battle, Rebel spies managed to steal secret plans to the Empire’s ultimate weapon, the DEATH STAR, an armored space station with enough power to destroy an entire planet.
        </p>
        <p>
          Pursued by the Empire’s sinister agents, Princess Leia races home aboard her starship, custodian of the stolen plans that can save her people and restore freedom to the galaxy….
        </p>
      </div>
    </section>
  </div>
</section>
`;
